Implementation of vkconfig doc html|settings|markdown <layername>

davidlunarg opened the pull request:
This is an initial implementation if adding "doc html|settings <layername>" to the command line options of vkconfig.

May still want to make further changes to add this capability to the GUI, and may want to change the output so as to better integrate the generation of layer documentation files with documentation needed for SDK creation.

ci-tester-lunarg commented:
CI VulkanTools build queued with queue ID 14313.

ci-tester-lunarg commented:
CI VulkanTools build # 1931 running.

ci-tester-lunarg commented:
CI VulkanTools build # 1931 passed.

ci-tester-lunarg commented:
CI VulkanTools build queued with queue ID 14699.

ci-tester-lunarg commented:
CI VulkanTools build # 1932 running.

ci-tester-lunarg commented:
CI VulkanTools build # 1932 passed.

ci-tester-lunarg commented:
CI VulkanTools build queued with queue ID 14744.

ci-tester-lunarg commented:
CI VulkanTools build # 1934 running.

ci-tester-lunarg commented:
CI VulkanTools build # 1934 passed.

christophe-lunarg commented on vkconfig_core/override.cpp:
```diff
@@ -182,19 +182,47 @@ bool WriteSettingsOverride(const Environment& environment, const std::vector<Lay
+
+            // If feature has unmet dependency, output it but comment it out
+            if (!::CheckDependence(*meta, parameter.settings)) {
+                stream << "#";
```
Nice!

christophe-lunarg reviewed (commented):
This is really great, I'll pull the branch to do further testing but your implementation is spendid.

christophe-lunarg commented:
```
$ ./vkconfig.exe doc
Missing 'doc' command argument...

Name
        'doc_html' - Command to create Vulkan layer doc files

Synopsis
        vkconfig doc html <layer_name>
        vkconfig doc settings <layer_name>

Description
        vkconfig doc html <layer_name>
                Create the html documentation file for the given layer.

        vkconfig doc settings <layer_name>
                Create the vk_layers_settings.txt file for the given layer.
```

ci-tester-lunarg commented:
CI VulkanTools build queued with queue ID 4046.

ci-tester-lunarg commented:
CI VulkanTools build # 1948 running.

davidlunarg commented:
Added commit 373b978.

Fixed doc_html in help text, fixed seg faults when not enough args to doc command, changed default location of output files to cwd, and added an optional 4th arg that specifies output dir.

The problem with:

`$ ./vkconfig.exe doc settings VK_LAYER_KHRONOS_validation
vkconfig: could not find layer VK_LAYER_KHRONOS_validation`

is that in order to create the settings file, the layer needs to be loaded. If vkconfig can't load the layer, it outputs an error message. I changed the error message to indicated that the layer could not be loaded.

ci-tester-lunarg commented:
CI VulkanTools build # 1948 passed.

ci-tester-lunarg commented:
CI VulkanTools build queued with queue ID 4434.

ci-tester-lunarg commented:
CI VulkanTools build # 1949 running.

ci-tester-lunarg commented:
CI VulkanTools build # 1949 passed.

ci-tester-lunarg commented:
CI VulkanTools build queued with queue ID 14434.

ci-tester-lunarg commented:
CI VulkanTools build # 1986 running.

ci-tester-lunarg commented:
CI VulkanTools build # 1986 passed.

ci-tester-lunarg commented:
CI VulkanTools build queued with queue ID 19622.

ci-tester-lunarg commented:
CI VulkanTools build # 2001 running.

ci-tester-lunarg commented:
CI VulkanTools build # 2001 passed.

davidlunarg commented:
Abandoning this pull request, will create a another.
